src/core-manager/index.js

```diff
@@ -478,6 +478,25 @@ export class CoreManager extends TypedEmitter {
     })
     return this.#corestore.replicate(stream)
   }
+
+  /**
+   * @param {Exclude<typeof NAMESPACES[number], 'auth'>} namespace
+   * @param {Object} [opts]
+   * @param {boolean} [opts.deleteOwn=false]
+   */
+  async deleteData(namespace, { deleteOwn = false } = {}) {
+    const coreRecords = this.getCores(namespace)
+    const ownWriterCore = this.getWriterCore(namespace)
+
+    const deletionPromises = []
+
+    for (const { core, key } of coreRecords) {
+      if (!deleteOwn && key === ownWriterCore.key) continue
```

Comment:
This is risky - I _think_ the `key` from coreRecords is the same instance of the key Buffer as the one used for the core instance, but Hypercore could change their implementation and it could be a copy.

```suggestion
      if (!deleteOwn && key.equal(ownWriterCore.key)) continue
```

Reply:
ah yes, forgot that this is a buffer instance and not a string - thanks for spotting!

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/419/commits/afa53f35344465881c4e2a4e0677a28e3b8ec063